.github/workflows/test.yml

```diff
@@ -0,0 +1,47 @@
+name: Test
+
+on:
+  workflow_run:
+    workflows:
+      - dynamoAllNet6.0_build.yml
+    types:
+      - completed
+
+env:
+  TEST_EXECUTE_DOWNLOAD_URL: https://downloads.smartbear.com/TestExecute1552SLM.exe
+
+jobs:
+  ui_smoke_tests:
+    name: UI Smoke tests
+    timeout-minutes: 60
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+            repository: DynamoDS/DynamoTestCompleteTests
+            token: ${{ sectrets.DYNAMO_TESTS_TOKEN }}
+      - name: Download TestExecute
+        run: Invoke-WebRequest -Uri "${{ env.TEST_EXECUTE_DOWNLOAD_URL }}" -OutFile ".\TE.exe"
+      - name: Install TestExecute
+        shell: cmd
+        run: .\TE.exe -SilentInstall
+      - name: Download Artifact
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: ${{ github.event.workflow_run.workflow_id }}
```

Comment:
is it possible that this code could be malicious - ie the artifact or workid could be controlled by a malicious user?

Reply:
The action source code is open https://github.com/dawidd6/action-download-artifact. It is downloading artifact from build workflow, which is net6 build of DynamoCoreRuntime which is open to download from github web interface as well. `workflow_id` is extracted from `github` context.